File: training/csharp/src/De.Sots.CellarsAndCentaurs/Application/CreatureUtil.cs
using De.Sots.CellarsAndCentaurs.Domain.Model;
using static De.Sots.CellarsAndCentaurs.Domain.Model.Dice;
using Fightable = De.Sots.CellarsAndCentaurs.Domain.Model.Fightable;

namespace De.Sots.CellarsAndCentaurs.Application;

public static class CreatureUtil
{
    public const string STANDARD_ARMOR_CLASS_DESCRIPTION = "Natural Armor";

    /*
     * Counts the treasure hoard a creature guards.
     */
    public static int TreasureHoard(Creature creature)
    {
        var d20Roll = RollD20();
        var xpValue = XPValue.ForCreature(creature);
        return d20Roll * xpValue.Points;
    }

    public static bool IsCentaur(Creature creature) => creature is Centaur;
}

File: training/ruby/lib/de/sots/cellars_and_centaurs/application/creature_util.rb
# frozen_string_literal: true

require_relative '../version'
require_relative '../domain/model/creature'
require_relative '../domain/model/fightable'
require_relative '../domain/model/dice'

module De::Sots::CellarsAndCentaurs::Application
  module CreatureUtil
    include De::Sots::CellarsAndCentaurs::Domain::Model

    STANDARD_ARMOR_CLASS_DESCRIPTION = 'Natural Armor'

=begin
Counts the treasure hoard a creature guards.
=end
    def self.hoard_size(creature)
      D20.roll.value * creature.xp_value
    end

    def self.describe(creature)
      "#{creature.type.to_s.capitalize} with #{creature.armor_class.description}"
    end
  end
end
